posthog/temporal/batch_exports/utils.py

```diff
@@ -137,24 +137,28 @@ def as_py(self) -> dict | None:
             if not value:
                 return None
 
+            json_bytes = value.encode("utf-8")
+
             try:
-                return orjson.loads(value.encode("utf-8"))
+                return orjson.loads(json_bytes)
             except orjson.JSONDecodeError:
                 pass
 
+            json_bytes = value.encode("unicode-escape").decode("utf-8", "replace").encode("utf-8")
```

Comment:
This round-trip takes care of escaping any invalid characters, like control characters (`\n`, `\t`) or unpaired UTF-16 surrogates.